src/CtaButton/CtaButton.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ctaButtonSizes, ctaButtonVariants } from './CtaButton.levers';
+import { iconOptions, iconAnimations } from "../common/icon/icon.levers";
+import getIconOptions from "../common/icon/getIconOptions";
+import getIconClasses from "../common/icon/getIconClasses";
+import getIconAnimation from "../common/icon/getIconAnimation";
+import Icon from 'react-hero-icon';
+import clsxd from 'clsx-dedupe';
+
+/**
+ * CTA Link Button Component
+ *
+ */
+export const CtaButton = ({ classes = {}, text, srText, link, variant, size, icon, animate, ...props }) => {
+  // Defaults & Variables.
+  // ---------------------------------------------------------------------------
+  const levers = {};
+
+  // Levers
+  // ---------------------------------------------------------------------------
+
+  // variant
+  if (variant && ctaButtonVariants.includes(variant)) {
+    switch (variant) {
+      case 'solid':
+        levers.variant = 'su-bg-digital-red hocus:su-bg-archway-dark su-text-white hocus:su-text-white su-border-2 su-border-digital-red su-border-solid hover:su-border-black focus:su-border-black';
+        break;
+
+      case 'outline':
+        levers.variant = 'su-bg-white hocus:su-bg-white su-text-digital-red hocus:su-text-black su-border-2 su-border-digital-red su-border-solid hover:su-border-black focus:su-border-black';
+        break;
+
+      case 'ghost':
+        levers.variant = 'su-bg-transparent su-text-white hocus:su-text-white su-border-2 su-border-white su-border-solid';
+        break;
+    }
+  }
+
+  // size
+  if (size && ctaButtonSizes.includes(size)) {
+    switch (size) {
+      case 'default':
+        levers.size = 'su-px-26 su-pt-10 su-pb-11 su-text-16 md:su-text-20';
+        break;
+
+      case 'big':
+        levers.size = 'su-px-34 su-py-15 su-text-20 md:su-text-24';
+        break;
+
+      case 'small':
+        levers.size = 'su-px-19 su-py-9 su-text-16 md:su-text-18';
+        break;
+    }
+  }
+
+  // icon
+  let heroicon = '';
+
+  if (icon && iconOptions.includes(icon)) {
+    heroicon = getIconOptions(icon);
+    levers.icon = getIconClasses(icon);
+  }
+
+  // animate
+
+  // Add specific classes for each type of animation
+  if (animate && iconAnimations.includes(animate)) {
+    levers.animate = getIconAnimation(animate);
+  }
```

Comment:
@sherakama Trying new approach. Is it worth doing this you think?

Reply:
Since we're separating out the CTA Link and Link Button components, we could either duplicate the icon/animation code and have them in each file, or have them shared this way. The duplicate way is more readable, but this way has less redundant code (there might be a better way to structure this).

Reply:
Also, more levers/values are sharable this way, e.g., the button sizes and variants are sharable with the CTA button links.